docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,240 @@
+- *A message ID* &ndash; A property used to distinguish messages.
+	It is useful for short messages where a collision might occur, like `%0 images`.
+- *A message string* &ndash; The default (English) form of the message.
+	When the message is supposed to support plural versions, this is the default singular version.
+- *A message plural* &ndash; An optional plural (English) version of the message.
+	The presence of this property indicates that the message should support both singular and plural forms.
+- *A PO file (`.po`)* &ndash; A file containing translations for the language in the [PO format](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html).
+	All CKEditor 5 localizable packages contain such files in the `lang/translations/` directory.
+- *A POT file (`.pot`)* &ndash; A file containing source messages (English sentences) that will be translated.
+- *A translation asset* &ndash; A JavaScript file or part of the file with generated translations for one language.
+
+## Writing a localizable UI
+
+All *messages* that need localization should be passed to the special CKEditor 5's {@link module:utils/locale~Locale#t `t()` function}. This function can be retrieved from the editor's {@link module:utils/locale~Locale} instance: `const { t } = editor.locale;` or from any view method: `const t = this.t;`.
+
+As the first argument, the `t()` function accepts either a string literal or an object literal containing the `id`, `string` and `plural` (optional) properties. The string literal will serve as both the *message ID* and the *message string*.
+
+As the second argument, the translation function accepts a value or an array of values. These values will be used to fill the placeholders in more advanced translation scenarios. If the `plural` property is specified, the first value will be used as the quantity determining the plural form.
+
+When using the `t()` function, you can create your own *localizable messages* or reuse *messages* created in CKEditor 5 packages that your project depends on. In case of reusing *messages*, you will not need to worry about translating them as all work will be done by the CKEditor 5 team and [Transifex translators](https://www.transifex.com/ckeditor/ckeditor5/) anyway. Obviously, your help in translating will still be appreciated!
+
+For simple *localizable messages*, use the string form for simplicity:
+
+```js
+const emojiName = 'cat';
+
+// Assuming that the English language was picked:
+t( 'insert emoji' ); // "insert emoji"
+t( 'insert %0 emoji', emojiName ); // "insert cat emoji"
+t( 'insert %0 emoji', [ emojiName ] ); // "insert cat emoji"
+```
+
+For more advanced scenarios, use plain object forms:
+
+```js
+const quantity = 3;
+
+// Assuming that the English language was picked:
+t( { string: '%0 emoji', id: 'INSERT_EMOJI' }, 'insert' ); // "insert emoji"
+t( { string: '%0 emoji', plural: '%0 emojis', id: 'EMOJI' }, quantity ); // "3 emojis"
```

Comment:
```suggestion
t( { string: '%0 emoji', plural: '%0 emojis', id: 'INSERT_EMOJI' }, quantity ); // "3 emojis"
```

Reply:
These are actually three different examples. IDK if it makes sense to have the same ids here.

Reply:
Hmm, I understand. But in this case, I'd propose other ids: EMOJI, N\_EMOJIS, ACTION\_N\_EMOJIS? How about that? :thinking: